lib/rules/require-named-unassigned-functions.js

```diff
@@ -0,0 +1,84 @@
+var assert = require('assert');
+var pathval = require('pathval');
+
+function getNodeName(node) {
+    if (node.type === 'Identifier') {
+        return node.name;
+    } else {
+        return node.value;
+    }
+}
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(requireNamedUnassignedFunctions) {
+        assert(
+            requireNamedUnassignedFunctions === true ||
+            typeof requireNamedUnassignedFunctions === 'object',
+            'requireNamedUnassignedFunctions option requires true value ' +
+            'or an object with String[] `allExcept` property'
+        );
+
+        // verify first item in `allExcept` property in object (if it's an object)
+        assert(
+            typeof requireNamedUnassignedFunctions !== 'object' ||
+            Array.isArray(requireNamedUnassignedFunctions.allExcept) &&
+            typeof requireNamedUnassignedFunctions.allExcept[0] === 'string',
+            'Property `allExcept` in requireNamedUnassignedFunctions should be an array of strings'
+        );
+
+        if (requireNamedUnassignedFunctions.allExcept) {
+            this._allExceptItems = requireNamedUnassignedFunctions.allExcept.map(function(item) {
+                var parts = pathval.parse(item).map(function extractPart (part) {
+                    return part.i !== undefined ? part.i : part.p;
+                });
+                return JSON.stringify(parts);
+            });
+        }
+    },
+
+    getOptionName: function() {
+        return 'requireNamedUnassignedFunctions';
+    },
+
+    check: function(file, errors) {
+        var _this = this;
+        file.iterateNodesByType(['FunctionExpression'], function(node) {
+            var parentNode = node.parentNode;
+            if (!parentNode.type.match(/VariableDeclarator|Property|AssignmentExpression/)) {
+                if (node.id === null) {
```

Comment:
`VariableDeclarator` nodes uses `id` property, `Property` uses `key`, `AssignmentExpression` uses `left`.

Absolutely not clear why we use here only `id` to check. ;-(

Can we use this structure?
```
// skip because...
if (parentNode.type.match...) {
  return;
}

// skip because
if (node.id === null) {
  return;
}

// ... the rest part of code
```